sdk/tests/features/unit/test_feature_extractor.py

```diff
@@ -0,0 +1,76 @@
+import unittest
+
+from features.feature_extractor import extract_all_features_from_package
+
+
+class FeatureExtractorTests(unittest.TestCase):
+    def test_get_valid_decorators(self):
+        package = "tests.features.unit.dummyfiles.valid"
+        expected_features = ["F4", "F3", "F1", "F2"]
+
+        feature_list_from_function = extract_all_features_from_package(package)
+        expected_set = set(expected_features)
+        actual_set = set(feature_list_from_function)
+
+        self.assertEqual(len(feature_list_from_function), 4)
+        self.assertEqual(expected_set, actual_set)
+
+    def test_if_throws_error_on_duplicate_decorators(self):
```

Comment:
you can reduce the code duplication by using the `parameterized` package see https://github.com/finbourne/lusid-sdk-python-preview/blob/master/sdk/tests/utilities/test_api_client_builder.py#L19 for an example